python/ray/serve/tests/test_regression.py

```diff
@@ -35,6 +38,26 @@ async def __call__(self, _request):
     assert result.json() == 100.0
 
 
+def test_backend_worker_memory_growth(serve_instance):
+    # https://github.com/ray-project/ray/issues/12395
+    client = serve_instance
+
+    def gc_tracked_objects(flask_request):
+        return len(gc.get_objects())
+
+    client.create_backend("model", gc_tracked_objects)
+    client.create_endpoint("model", backend="model", route="/model")
+
+    tracked_objects: List[int] = []
+    for _ in range(10):
+        result = requests.get("http://127.0.0.1:8000/model")
+        assert result.status_code == 200
+        tracked_objects.append(result.json())
+
+    # gc tracked object shouldn't grow
+    assert tracked_objects[5] == tracked_objects[-1]
```

Comment:
This looks somewhat brittle, let's just check that there aren't any unreachable references (that's what we really care about):

```
``` 
import gc
gc.set_debug(gc.DEBUG_SAVEALL)
gc.collect()
assert len(gc.garbage) == 0
```